examples/svdiff/modeling_svdiff.py

```diff
@@ -0,0 +1,133 @@
+        """
+        super().__init__()
+        self.weight_type = weight_type
+        weight = self._reshape_weight_for_svd(weight)
+        # perform SVD
+        with torch.no_grad():
+            U, S, Vh = torch.linalg.svd(weight, full_matrices=False)
+        self.register_buffer("U", U.detach())
+        self.U.requires_grad = False
+        self.register_buffer("S", S.detach())
+        self.S.requires_grad = False
+        self.register_buffer("Vh", Vh.detach())
+        self.Vh.requires_grad = False
+        # initialize to 0 for smooth tuning 
+        self.delta = nn.Parameter(torch.zeros_like(S)) if init_delta is None else nn.Parameter(init_delta)
+        self.scale = scale
+
+    def _reshape_weight_for_svd(self, updated_weight: torch.Tensor, original_weight: torch.Tensor = None, reverse: bool = False, **kwargs):
+        if self.weight_type is not None:
+            if self.weight_type == "conv":
+                if not reverse:
+                    shape = updated_weight.shape
+                    return torch.reshape(updated_weight, (shape[0], shape[1:].numel()))
+                else:
+                    return torch.reshape(updated_weight, original_weight.shape)
+            elif self.weight_type == "1d":
+                if not reverse:
+                    return updated_weight.unsqueeze(0)
+                else:
+                    return updated_weight.squeeze(0)
+            else:
+                raise ValueError(f"`weight_type`={self.weight_type} is invalid!")
+        else:
+            return updated_weight
+        
+    def forward(self, weight: torch.Tensor):
+        updated_weight = self.U @ torch.diag(F.relu(self.S + self.scale * self.delta)) @ self.Vh
+        updated_weight = self._reshape_weight_for_svd(updated_weight, weight, reverse=True)
+        return updated_weight
+
+
+def set_spectral_shifts(model: nn.Module, spectral_shifts_ckpt: str = None, **kwargs):
+    # key to module
+    svdiff_modules = {}
+    for name, module in model.named_modules():
+        svdiff_module = None
+        if isinstance(module, nn.Conv1d) or isinstance(module, nn.Conv2d):
+            svdiff_module = SVDiffModule(module.weight, weight_type="conv")
+        elif isinstance(module, nn.LayerNorm) or isinstance(module, nn.GroupNorm):
+            svdiff_module = SVDiffModule(module.weight, weight_type="1d")
+        elif isinstance(module, nn.Linear) or isinstance(module, nn.Embedding):
+            svdiff_module = SVDiffModule(module.weight)
+        if svdiff_module:
+            svdiff_modules[name+".delta"] = svdiff_module
+            # register parametrization 
+            nn.utils.parametrize.register_parametrization(module, "weight", svdiff_module)
+            
+    if spectral_shifts_ckpt:
+        param_device = "cpu"
+        torch_dtype = kwargs["torch_dtype"] if "torch_dtype" in kwargs else None
+        with safetensors.safe_open(spectral_shifts_ckpt, framework="pt", device="cpu") as f:
```

Comment:
I think it's better to allow the standard `.bin` here as well to maintain consistency with the rest of our example.

Reply:
In train_svdiff.py, you only get safetensors files for spectral shifts. https://github.com/huggingface/diffusers/blob/6a811c24f86bc97f863e923f6264045adf27da23/examples/svdiff/train_svdiff.py#L907

So, it doesn't need to handle `.bin`.

Reply:
Okay for me. Cc: @patrickvonplaten